src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -411,7 +418,7 @@ private Exception WrapExceptionForModelState(Exception exception)
             // we regard as having safe messages to expose to clients
             var isJsonExceptionType =
                 exception is JsonReaderException || exception is JsonSerializationException;
-            var suppressJsonDeserializationExceptionMessages = _options?.SuppressJsonDeserializationExceptionMessagesInModelState ?? _suppressJsonDeserializationExceptionMessages;
+            var suppressJsonDeserializationExceptionMessages = _jsonOptions?.AllowInputFormatterExceptionMessages ?? _allowInputFormatterExceptionMessages;
```

Comment:
Should this be `!_jsonOptions?.AllowInputFormatterExceptionMessages`

Reply:
Yeah I think you're right. durr

Reply:
I'm cleaning up this code in general, it's a little hard to grok.